store/index-refactor.js

```diff
@@ -168,13 +168,17 @@ export const mutations = {
 
     initializeDataDictionary(p_state) {
 
+        // 1. Create a skeleton data dictionary based on the data table's columns
         let dataDictionary = {};
-
         for ( const columnName of Object.keys(p_state.dataTable[0]) ) {
 
             dataDictionary[columnName] = {"description": ""};
         }
 
+        // 2. Create a copy of the data dictionary to save as an original
+        p_state.dataDictionary.provided = Object.assign({}, dataDictionary);
```

Comment:
Good catch about the `provided`, but I think it should go with `setDataDictionary`, no? Currently the dictionary that the user supplies via `processDataDictionary` -> `setDataDictionary` is never stored in `p_state.dataDictionary.provided`. 

I don't disagree with `.provided` also being writter to here, but it definitely must be written to in `setDataDictionary` when the user provided datadictionary is parse.

Reply:
Yeah. That's slicker. I see what you mean. Will commit the change in a minute.